src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -2,13 +2,46 @@
 
 import core.basesyntax.dao.StorageDao;
 import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.db.Storage;
+import core.basesyntax.exception.RegisterException;
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
+    private static final int MIN_PASSWORD_AND_LOGIN_LENGTH = 6;
+    private static final int MIN_AGE = 18;
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
     public User register(User user) {
-        return null;
+
+        if (user == null) {
+            throw new RegisterException("User can not be null");
+        }
+
+        for (User a : Storage.people) {
+            if (a.getLogin().equals(user.getLogin())) {
+                throw new RegisterException("Login already exists! Enter another value");
+            }
+        }
+        if (user.getLogin() == null) {
+            throw new RegisterException("Login can not be null!");
+        }
```

Comment:
The check for `user.getLogin() == null` should be performed before checking if the login already exists in the storage. This will prevent a `NullPointerException` when calling `a.getLogin().equals(user.getLogin())` if `user.getLogin()` is null.

Reply:
fixed